README.md

```diff
@@ -36,38 +36,39 @@ Run `make help` for a list of target options. Here are some of them:
 
 ```
 Cleaning targets:
-  clean                      - clean the cache artifacts
+  clean                      - clean the cache artifacts and generated files
 Generic targets:
-* all                        - build solidity code. To build from a clean clone, run: make submodules downloads all
+* all                        - build solidity code. To build from a clean clone, run: make submodules all
   build                      - build solidity code
-  dep                        - install npm packages
-  depclean                   - remove npm packages
-  deploy                     - deploy to local node
-  generate                   - generate solidity code from cpp and template
-  test                       - test both binary files and log files
-  coverage                   - generate coverage report for html view
+  generate-all               - generate all solidity code
+  generate-step              - generate solidity-step code from cpp
+  generate-mock              - generate mock library code
+  generate-prod              - generate production library code
+  generate-replay            - generate replay tests
+  pretest                    - download necessary files for tests
+  test-all                   - test all
+  test-mock                  - test binary files with mock library
+  test-prod                  - test production code
+  test-replay                - test log files
 ```
 
-Obs: Some make targets require the installation of the node.js packages. This can be done with `yarn install` or `make dep`.
+### Prerequisite
+
+-   Build [Cartesi docker image](https://github.com/cartesi/machine-emulator#getting-started)
 
 ### Requirements
 
 -   Foundry 0.2.0
 -   GNU Make >= 3.81
--   Lua 5.3.5
--   Yarn 1.22.19
+-   GPP >= 2.27
```

Comment:
I tried `cpp`, however it lacks some feature flexibility that `gpp` offers. This can be a future improvement but I am afraid we don't have time to change this for this release.